docs/guides/code/min_max_concurrency.py

```diff
@@ -0,0 +1,16 @@
+from crawlee import ConcurrencySettings
+from crawlee.beautifulsoup_crawler import BeautifulSoupCrawler
+
+concurrency_settings = ConcurrencySettings(
+    # Start the crawler right away with 8 concurrent tasks, if they are available
+    desired_concurrency=8,
+    # Ensure there will always be minimum 5 concurrent tasks at any time
+    min_concurrency=5,
+    # Ensure the crawler doesn't exceed 10 concurrent tasks at any time
+    max_concurrency=10,
+)
+
+crawler = BeautifulSoupCrawler(
+    # Pass the concurrency setting to the crawler
+    concurrency_settings=concurrency_settings,
+)
```

Comment:
```suggestion
import asyncio

from crawlee import ConcurrencySettings
from crawlee.beautifulsoup_crawler import BeautifulSoupCrawler


async def main() -> None:
    concurrency_settings = ConcurrencySettings(
        # Start with 8 concurrent tasks, as long as resources are available.
        desired_concurrency=8,
        # Maintain a minimum of 5 concurrent tasks to ensure steady crawling.
        min_concurrency=5,
        # Limit the maximum number of concurrent tasks to 10 to prevent
        # overloading the system.
        max_concurrency=10,
    )

    crawler = BeautifulSoupCrawler(
        # Use the configured concurrency settings for the crawler.
        concurrency_settings=concurrency_settings,
    )


if __name__ == '__main__':
    asyncio.run(main())
```